TTS/tts/layers/losses.py

```diff
@@ -585,6 +586,11 @@ def kl_loss(z_p, logs_q, m_p, logs_p, z_mask):
         l = kl / torch.sum(z_mask)
         return l
 
+    @staticmethod
+    def cosine_similarity_loss(gt_spk_emb, syn_spk_emb):
+        l = - torch.nn.functional.cosine_similarity(gt_spk_emb, syn_spk_emb).mean()
+        return l
+
     def forward(
```

Comment:
Optionally would be nice to have a test for the loss. It is quite a complex loss layer :)